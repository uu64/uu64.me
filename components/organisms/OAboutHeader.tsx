import React from "react";
import blog from "@/data/blog.json";

const OAboutHeader: React.FC = () => {
  return (
    <div className="py-8">
      <h1 className="font-semibold text-3xl text-custom-normal">
        {blog.pages.about.title}
      </h1>
      <p className="border-b-2 border-gray-400 py-2 text-sm">
        {blog.pages.about.description}
      </p>
    </div>
  );
};

export default OAboutHeader;
