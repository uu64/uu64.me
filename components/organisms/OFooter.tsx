import React from "react";
import siteconfig from "@/siteconfig.json";

const OFooter: React.FC = () => {
  return (
    <>
      <svg
        className="pt-12 fill-night1"
        version="1.1"
        viewBox="0 0 1228 80.448"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path d="m0.096151 25.992c182.86-27.18 266.83-25.886 345.19-25.886s433.22 32.243 543.82 45.3c175.4 20.709 338.76-1.2943 338.76-1.2943v36.24h-1227.8z" />
      </svg>
      <div className="bg-custom-night1 text-center pt-4 pb-8">
        {siteconfig.copyright}
      </div>
    </>
  );
};

export default OFooter;
