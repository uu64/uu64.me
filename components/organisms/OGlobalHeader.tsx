import React from "react";
import ALogo from "@/components/atoms/ALogo";
import MMenu from "@/components/molecules/MMenu";

const OGlobalHeader: React.FC = () => {
  return (
    <nav>
      <div className="max-w-2xl px-8 py-4 flex items-end mx-auto">
        <div className="flex-grow">
          <ALogo />
        </div>
        <MMenu />
      </div>
    </nav>
  );
};

export default OGlobalHeader;
