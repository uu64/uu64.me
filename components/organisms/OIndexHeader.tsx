import React from "react";
import siteconfig from "@/siteconfig.json";

const OIndexHeader: React.FC = () => {
  return (
    <div className="py-8">
      <h1 className="font-semibold text-3xl text-custom-normal">
        {siteconfig.pages.blog.title}
      </h1>
      <p className="border-b-2 border-gray-400 py-2 text-sm">
        {siteconfig.pages.blog.description}
      </p>
    </div>
  );
};

export default OIndexHeader;
