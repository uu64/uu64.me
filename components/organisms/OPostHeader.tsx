import React from "react";

interface Props {
  frontMatter: FrontMatter;
}

const OPostHeader: React.FC<Props> = (props: Props) => {
  const { frontMatter } = props;
  return (
    <div className="py-8">
      <div>{frontMatter.date}</div>
      <h1 className="border-b-2 border-gray-400 font-semibold text-custom-normal py-2">
        {frontMatter.title}
      </h1>
    </div>
  );
};

export default OPostHeader;
