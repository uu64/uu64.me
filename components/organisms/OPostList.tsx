import React from "react";
import MPost, { IPostItem } from "@/components/molecules/MPost";

interface Props {
  posts: IPostItem[];
}

const OPostList: React.FC<Props> = (props: Props) => {
  const { posts } = props;
  return (
    <ul>
      {posts.map((post) => (
        <li key={post.resourcePath}>
          <MPost post={post} />
        </li>
      ))}
    </ul>
  );
};

export default OPostList;
