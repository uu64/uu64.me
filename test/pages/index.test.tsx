import React from "react";
import renderer from "react-test-renderer";
import mdxUtil from "@/lib/mdx-util";
import { IPostItem } from "@/components/organisms/OPostList";
import App from "@/pages/index";

describe("App", () => {
  it("renders correctly", async () => {
    const posts = await mdxUtil.getPosts();
    const component = renderer.create(
      <App
        // Use the 5 oldest posts
        posts={posts.slice(-5).map(
          (post): IPostItem => {
            return {
              resourceId: post.resourceId,
              date: post.frontMatter.date,
              title: post.frontMatter.title,
            };
          }
        )}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
