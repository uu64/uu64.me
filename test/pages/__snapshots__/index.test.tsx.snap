// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="blog-index-content px-8"
>
  <div
    className="py-8"
  >
    <h1
      className="font-semibold text-3xl text-custom-normal"
    >
      uu64.me
    </h1>
    <p
      className="border-b-2 border-gray-400 py-2 text-sm"
    >
      技術的な活動の記録など。
    </p>
  </div>
  <ul>
    <li>
      <div
        className="py-4"
      >
        <p
          className="py-2"
        >
          2020-07-19
        </p>
        <a
          className="font-semibold text-custom-normal"
          href="/blog/2020/07/unix-philosophy"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <h2>
            『UNIXという考え方 その設計思想と哲学』を読んで
          </h2>
        </a>
      </div>
    </li>
    <li>
      <div
        className="py-4"
      >
        <p
          className="py-2"
        >
          2020-07-04
        </p>
        <a
          className="font-semibold text-custom-normal"
          href="/blog/2020/07/renew-blog"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <h2>
            Next.js + TypeScript + MDXでブログシステムを構築した
          </h2>
        </a>
      </div>
    </li>
    <li>
      <div
        className="py-4"
      >
        <p
          className="py-2"
        >
          2020-05-13
        </p>
        <a
          className="font-semibold text-custom-normal"
          href="/blog/2020/05/serverless-singlepage-application"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <h2>
            O'REILLY サーバーレスシングルページアプリケーション を読んで
          </h2>
        </a>
      </div>
    </li>
    <li>
      <div
        className="py-4"
      >
        <p
          className="py-2"
        >
          2019-08-27
        </p>
        <a
          className="font-semibold text-custom-normal"
          href="/blog/2019/08/ionic-tutorial"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <h2>
            Ionicのチュートリアルをやってみた
          </h2>
        </a>
      </div>
    </li>
    <li>
      <div
        className="py-4"
      >
        <p
          className="py-2"
        >
          2019-08-13
        </p>
        <a
          className="font-semibold text-custom-normal"
          href="/blog/2019/08/new-blog"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <h2>
            ブログを引っ越しました
          </h2>
        </a>
      </div>
    </li>
  </ul>
</div>
`;
